### PR TITLE
[BEAM-746] Choose BigQuery Write implementation based on Input PCollection

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

Stop using PipelineOptions, and instead use the boundedness of the input
to choose how to write to BigQuery. This means that runners that don't
use the streaming flag still function appropriately.

Fixes BEAM-746

### CODE REVIEW

**Comment by tgroh** [comment]
R: @peihe CC @dhalperi

**Comment by peihe** [comment]
Have you verified the PR with some bounded big query tests in streaming?

**Comment by tgroh** [comment]
Yes; I've ran `WindowedWordCount` on the `DataflowRunner` and set `--streaming`; the BigQuery table was loaded

**Comment by peihe** [comment]
I think we can verify more than table was loaded.

There is big query io coder unbounded read tests. You can port the one line change and run the test 20 times easily.
The test will verify the output is expected, and also tests if it is flaky.

Even though this is just one line, it switches to new code path. And, bounded batch write sometimes doesn't work automatically in streaming.

**Comment by peihe** [comment]
LGTM
(@tgroh verified with integration tests)

**Comment by tgroh** [comment]
R: @dhalperi for committer

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java [hunk 1]
```diff
@@ -972,12 +974,28 @@ private void testWriteValidatesDataset(boolean streaming) throws Exception {
     thrown.expectMessage(
         Matchers.either(Matchers.containsString("Unable to confirm BigQuery dataset presence"))
             .or(Matchers.containsString("BigQuery dataset not found for table")));
-    p.apply(Create.<TableRow>of().withCoder(TableRowJsonCoder.of()))
-     .apply(BigQueryIO.Write
-         .to(tableRef)
-         .withCreateDisposition(CreateDisposition.CREATE_IF_NEEDED)
-         .withSchema(new TableSchema())
-         .withTestServices(fakeBqServices));
+    PCollection<TableRow> tableRows;
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java]: please move all code that does not throw the exception above the `thrown.expect`... code.

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java [hunk 2]
```diff
@@ -1750,7 +1751,7 @@ public PDone apply(PCollection<TableRow> input) {
 
         // In a streaming job, or when a tablespec function is defined, we use StreamWithDeDup
```
Comment (dhalperi) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java]: command needs updating.

Does any javadoc need updating? please review.

Comment (tgroh) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java]: Updated. There's no other reference to streaming that's not related specifically to the BigQuery Streaming API

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java (hunk 1 above)
Comment (tgroh) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java]: Done.

**Comment by dhalperi** [comment]
LGTM

**Comment by markflyhigh** [comment]
Starting from this PR, WindowedWordCount post-commit test (streaming only) took double time than before (from 4.5mins to 10mins). It also slow down DirectRunner in pre-commit.

see: https://builds.apache.org/view/Beam/job/beam_PostCommit_MavenVerify/1621/org.apache.beam$beam-examples-java/testReport/org.apache.beam.examples/WindowedWordCountIT/

**Pull request closed without merging**